File: app/src/main/java/com/slq/tokfm/ByteRequest.java
package com.slq.tokfm;

import com.android.volley.NetworkResponse;
import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.HttpHeaderParser;

public class ByteRequest extends Request<byte[]> {

    private Response.Listener<byte[]> responseListener;

    public ByteRequest(String url, Response.Listener<byte[]> responseListener) {
        super(Method.GET, url, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                error.printStackTrace();
            }
        });
        this.responseListener = responseListener;
    }

    @Override
    protected Response<byte[]> parseNetworkResponse(NetworkResponse response) {
        return Response.success(response.data, HttpHeaderParser.parseCacheHeaders(response));
    }

    @Override
    protected void deliverResponse(byte[] response) {
        responseListener.onResponse(response);
    }
}

File: app/src/main/java/com/slq/tokfm/ByteRequest.kt
package com.slq.tokfm

import com.android.volley.NetworkResponse
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.HttpHeaderParser

class ByteRequest(url: String, val responseListener: Response.Listener<ByteArray>)
    : Request<ByteArray>(Request.Method.GET, url, Response.ErrorListener { error -> error?.printStackTrace() }) {

    override fun deliverResponse(response: ByteArray) = responseListener.onResponse(response)

    override fun parseNetworkResponse(response: NetworkResponse): Response<ByteArray> {
        return Response.success(response.data, HttpHeaderParser.parseCacheHeaders(response))
    }
}
